### DISCUSSION TITLE
Rebasing a temporary private fork of a repository after addressing a security advisory

### DISCUSSION THREAD

**Discussion by EolRes** [community/community#65923] [New to GitHub]
We have recently addressed a security advisory and in the process we created a temporary fork of a repository. How do we proceed to rebasing the fork? The original repo suffered changes after the fork, so I need to bring those changes to the private fork.

I could not find any documentation on this as well.

**Comment by lab596** [comment]
Here are the basic commands I found online: 

1. Clone Your Private Fork:
If you haven't already, clone your private fork of the repository to your local machine using the following command, replacing <your-fork-url> with the URL of your private fork:
`git clone <your-fork-url>`

2. Add the Original Repository as a Remote:
In your local clone of the private fork, you should add the original repository (the one you forked from) as a remote. You can do this with the following command, replacing <original-repo-url> with the URL of the original repository:
`git remote add upstream <original-repo-url>`

3. Fetch Changes from the Original Repository:
Fetch the latest changes from the original repository:
`git fetch upstream`

4. Rebase Your Fork:
Once you have fetched the latest changes from the original repository, you can rebase your private fork onto the updated upstream branch. Assuming you are working on the main branch, you can use the following command:
```
git checkout main
git rebase upstream/main
```
5. Push the Changes:
After the rebase is complete, push the changes to your private fork:
`git push origin main`
